src/Table/Table.js

```diff
@@ -24,13 +24,14 @@ class Table extends Component {
   }
 
   render() {
-    const { classes, className: classNameProp, children, ...other } = this.props;
+    const { classes, className: classNameProp, children, component, ...other } = this.props;
     const className = classNames(classes.root, classNameProp);
+    const ComponentTable = component || 'table';
```

Comment:
What about using a `defaultProps` instead? That's the pattern we try to follow everywhere. This has the advantage of being automatically documented.